Kim junho

xjvmdutl opened the pull request.

xjvmdutl committed:
commit -m Atdd SupplementRequest 속성 추가

xjvmdutl committed:
Merge branch 'mandykr:main' into KimJunho

xjvmdutl committed:
commit -m Atdd SupplementRequest 속성 추가

xjvmdutl committed:
commit -m SupplementContent, SupplementTitle 분리 작업

xjvmdutl committed:
commit -m SupplementContent, SupplementTitle 단위 테스트 작성

mandykr commented on src/test/java/com/example/nutrient/domain/SupplementContentTest.java:
```diff
@@ -0,0 +1,63 @@
+            () -> new SupplementContent(serialNumber, permitDate, expirationDate, intake,
+                mainFunctional, precautions, storageWay))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
```
테스트의 가독성이 조금 떨어지는것 같아요.
데이터의 크기를 줄이고, 영양제 내용안에 속한 필드별로 테스트를 분리하는건 어떻게 생각하시나요?

mandykr commented on src/main/java/com/example/nutrient/domain/SupplementContent.java:
```diff
@@ -0,0 +1,130 @@
+        expirationDateValidation(expirationDate);
+        intakeValidation(intake);
+        mainFunctionalValidation(mainFunctional);
+        precautionsValidation(precautions);
```
각 유효성 검사 호출하는 부분을 validate() 메소드로 먼저 분리하는건 어떨까요?

mandykr commented on src/main/java/com/example/nutrient/domain/SupplementContent.java:
```diff
@@ -0,0 +1,130 @@
+        this.mainFunctional = mainFunctional;
+        this.precautions = precautions;
+
+        this.storageWay = storageWay == null ? null : storageWay;
```
3항 연산을 사용하신 이유가 있으신가요?

mandykr commented on src/main/java/com/example/nutrient/domain/repository/jpa/SupplementJpaRepository.java:
```diff
@@ -0,0 +1,11 @@
+
+public interface SupplementJpaRepository extends SupplementRepository, JpaRepository<Supplement, UUID> {
+
+}
```
jpa에 의존적인 JpaRepository는 infra 패키지를 만들어 옮겨주셔도 좋을것 같아요

mandykr commented on src/test/java/com/example/nutrient/domain/SupplementContentTest.java:
```diff
@@ -0,0 +1,63 @@
+
+
+    @Test
+    @DisplayName("허용일자는 현재일 보다 작아야한다.")
```
허용일자와 현재일자가 같은 경우에 대한 테스트도 추가되면 좋을것 같아요.

mandykr commented on src/main/java/com/example/nutrient/application/dto/supplement/SupplementCreateResponse.java:
```diff
@@ -9,18 +14,40 @@
+    private String storageWay;
     private CategoryResponse category;
 
+    public static SupplementCreateResponse create(Supplement supplement) {
+        SupplementTitle title = supplement.getTitle();
+        SupplementContent content = supplement.getContent();
+        return new SupplementCreateResponse(
+            supplement.getId(),
+            title.getName(),
+            content.getSerialNumber(),
+            content.getPermitDate(),
+            content.getExpirationDate(),
+            content.getIntake(),
+            content.getMainFunctional(),
+            content.getPrecautions(),
+            content.getStorageWay(),
+            new CategoryResponse(UUID.randomUUID(), "임시 테스트")
+        );
+    }
```
VO객체를 파라미터로 받는것과
원시값을 파라미터로 받고 생성자 내에서 VO객체를 생성하는
두 가지 방법에 대해 함께 고민해보면 좋을것 같아요.

xjvmdutl committed:
commit -m 코드리뷰 수정 -> 허용 일자 = now 와 같을경우 테스트 케이스 추가, JPA에 의존적인 repository Infra로 이동

xjvmdutl committed:
commit -m UUID CUSTOM VALIDATION 추가

mandykr commented on src/main/java/com/example/nutrient/domain/SupplementContent.java:
```diff
@@ -0,0 +1,130 @@
+        this.storageWay = storageWay == null ? null : storageWay;
+    }
+
+    private void serialNumberValidation(String serialNumber) {
```
메소드명 수정이 필요해보입니다.
`validateSerialNumber`과 같이 동사형태로 변경하는게 어떨까요?

mandykr commented on src/main/java/com/example/nutrient/domain/SupplementContent.java:
```diff
@@ -0,0 +1,130 @@
+        }
+
+    }
+    private boolean isOverToNow(LocalDate permitDate) {
```
`영양제 내용`클래스가 갖는 `isOverToNow`메소드의 이름이 어색하게 느껴집니다.
`영양제 내용이 현재보다 큰가?`라고 해석될 수 있을 것 같아요.
메소드명에 `permitDate`가 포함 되도록 수정하면 좋을 것 같습니다.

xjvmdutl committed:
commit -m 코드 리뷰 수정 -> 메소드명 변경 / Validate 메소드 추출

xjvmdutl committed:
chore(SupplementContentTest): 테스트 예외 변경
- IllegalArgumentException -> SupplementException
chore(SupplementErrorCode): 예외 코드 추가
- SupplementErrorCode -> 예외를 Status,message 형식으로 관리한다.
chore(SupplementException): 예외 추가
- SupplementException 을 Supplement 에 관련된 예외 처리
chore(SupplementContent): 예외 변경
- IllegalArgumentException -> SupplementException

xjvmdutl committed:
chore(SupplementException): SupplementException 예외 발생(부모로 위임)
- Runtime 예외를 실행시켜 errorCode 메시지 전달

mandykr commented on src/main/java/com/example/nutrient/application/customvalidation/UUIDValidator.java:
```diff
@@ -0,0 +1,31 @@
+
+    private boolean isUUIDSize(String uuid) {
+        return uuid.length() == UUID_SIZE;
+    }
```
`isUUIDSize` 메소드명 변경이 필요해 보입니다.

mandykr commented on src/main/java/com/example/nutrient/application/SupplementService.java:
```diff
@@ -2,13 +2,28 @@
 
```
[커밋 메시지 규약](https://velog.io/@outstandingboy/Git-%EC%BB%A4%EB%B0%8B-%EB%A9%94%EC%8B%9C%EC%A7%80-%EA%B7%9C%EC%95%BD-%EC%A0%95%EB%A6%AC-the-AngularJS-commit-conventions)을 참고해주세요.

xjvmdutl committed:
chore(UUIDValidator): UUIDValidator 메소드 명 변경
- isUUIDSize -> isUUID
- 정규 표현식을 사용하여 UUID 형식을 체크한다
chore(UUIDValidatorTest): UUIDValidator 테스트 작성
- UUID 형식이 올바른지 확인하는 테스트 작성
